### PR TITLE
Add reducers

### PR DESCRIPTION
**Pull request by Lusindiso**
### In this PR I have made the following changes:
- Configure redux store using redux toolkit.
- Add books reducer with create slice,
- Add books add book and remove book action creators.
- Add category reducer with create slice,
- Add check status action creator.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    removeBook(state, action) {
+      // eslint-disable-next-line no-param-reassign
+      state.books = state.books.filter((book) => book.id !== action.payload.id);
+    },
+  },
+});
+
+export const booksActions = books.actions;
+
+export default books;
```
Comment (Bondok6) [inline src/redux/books/books.js:19]: - [ ] I can see that you used `createSlice` from `toolkit` to write your book's actions and reducers and this is not wrong, But at this point and according to the [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md), you should write your book's actions and reducer using the [ducks pattern](https://github.com/erikras/ducks-modular-redux) and the value of each action type needs to follow the ducks pattern. So kindly recreate this file using [ducks pattern](https://github.com/erikras/ducks-modular-redux) to follow the requirements 💯👌

> It should be something like

```
// Define Action Type
const ADD = 'bookstore/books/ADD';
const REMOVE = 'bookstore/books/REMOVE';

export default function booksReducer(state = [], action) {
  switch (action.type) {
    case ADD: return ... ;
    case REMOVE: return ... ;
    default: return state;
  }
}

export function addBook(book) { ... }

export function removeBook(id) { ... }
```

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  reducers: {
+    checkStatus() {
+      return 'Under construction';
+    },
+  },
+});
+
+export const categoriesActions = categories.actions;
+
+export default categories;
```
Comment (Bondok6) [inline src/redux/categories/categories.js:15]: - [ ] The same thing here, Please check the comment above ⬆

**Review by Bondok6 — CHANGES_REQUESTED** [review]
## Hi @Lusindiso 🙋‍♂️, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- No linter errors ✅
- Your repo is professional ✅
- Professional README file ✅

## Required Changes ♻️

- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categories.js (hunk 2 above)
Comment (Lusindiso) [reply src/redux/categories/categories.js:15]: Hi @Bondok6 
I think you are missing the point of ducks pattern, the requirement is that 
- It must export a reducer function,
- export an action creator
- MUST have action types in the form npm-module-or-app/reducer/ACTION_TYPE (which redux toolkit is an npm module)
All this has been followed hence the ducks pattern is followed.
You can also reference the [redux official documentation](https://redux.js.org/style-guide/) 
Which also recommends using redux toolkit and states clearly that it follows the ducks pattern. 
(You can just search for ducks pattern on the page to easily find the reference).

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @Lusindiso , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- No linter errors

 ## Required Changes ♻️

- [ ] Even though you implemented reducers, the functionality for adding and removing a book still doesn't work. It might be because you did not add the store yet to the application. Consider fixing this error.

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lusindiso** [comment]
@muneebulrehman 
That is not the requirement for the project, it is in fact the requirement for the following project.

**Comment by muneebulrehman** [comment]
> @muneebulrehman That is not the requirement for the project, it is in fact the requirement for the following project.
Hey @Lusindiso, if you feel that way, kindly ask for a second opinion.

**Review by Lameck1 — COMMENTED** [review]
This PR is under second opinion review, please hold on with any other activity.

**Review by Lameck1 — COMMENTED** [review]
Hello @Lusindiso 👋,

_This is an additional review due to the second opinion request._

I would like to confirm that your implementation of this milestone is on point. 

@Bondok6, createSlice automatically generates action creators and action types that correspond to the reducers and state. Therefore it will automatically generate actions using the ducks pattern.

@muneebulrehman, the required change is not for this milestone.

As this was the only reason your project was not approved it will be approved as soon as possible.

Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away :)

Keep rocking 🚀!

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @Lusindiso 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Lusindiso** [comment]
@Lameck1 Thank you for the clarification. @youmari thank you for the review.

**Pull request merged**